Packs/Cryptosim/Integrations/Cryptosim/Cryptosim.py

```diff
@@ -0,0 +1,266 @@
+
+    def connection_test(self):
+        return self._http_request("GET", data={}, url_suffix="correlations?limit=1")
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def correlation_alerts_command(client: Client):
+
+    # Call the Client function and get the raw response
+    result = client.correlation_alerts()
+    readable_data = []
+    for res in result["Data"]:
+        res = res["CorrelationAlert"]
+        readable_data.append(
+            {"ID": res.get('ID', ""), "CORRELATIONID": res.get('CORRELATIONID', ""),
+            "RULEID": res.get('RULEID', ""), "NAME": res.get('NAME', ""),
+            "Severity": res.get('RISK', ""),
+             "Created At": res.get('EVENTSTARTDATE', "")})
+    markdown = tableToMarkdown('Messages', readable_data, headers=['ID', 'CORRELATIONID', 'NAME', 'RULEID', 'Severity', 'Created At'])
+    return CommandResults(
+        outputs_prefix='CorrelationAlerts',
+        outputs_key_field='',
+        readable_output=markdown,
+        outputs=result,
+    )
+
+
+
+def correlations_command(client: Client):
+    result = client.correlations()
+
+
+    return CommandResults(
+        outputs_prefix='Correlations',
+        outputs_key_field='',
+        outputs=result,
+    )
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+    :type client: ``Client``
+    :param Client: client to use
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        if client.connection_test().get('StatusCode') == 200:
+            message = 'ok'
+        else:
+            raise Exception(f"""StatusCode:
+                            {client.correlations().get('StatusCode')},
+                            Error: {client.correlations().get('ErrorMessage')}
+                            """)
+    except DemistoException as e:
+        if '401' in str(e):
+            message = 'Authorization Error: make sure API User and Password is correctly set'
+        else:
+            raise e
+    return message
+
+
+''' INCIDENT '''
+
+
+def fetch_incidents(client: Client,params):
+
+    max_results = arg_to_number(arg=params.get('max_fetch'), arg_name='max_fetch', required=False)
+
+    first_fetch_time = arg_to_datetime(params.get('first_fetch'), "1 hour").strftime(DATE_FORMAT)
+
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('last_fetch', first_fetch_time)
+
+    incidentsList = []
+    alert_response = client.correlation_alerts(last_fetch_time=last_fetch)
+    incident_data = alert_response.get("Data",[])
+
+    for inc in incident_data:
+
+        if len(incidentsList) > max_results:
+            break
+
+        incident_name = inc['CorrelationAlert']['NAME']
+        time_stamp = inc['CorrelationAlert']['CREATEDATE'] + "Z"
```

Comment:
Didn't you say you are not getting it timezoned?

Reply:
'CREATEDATE' is local date, 
I create a instance params get time_zone and I added it to XSOAR's time values

Reply:
I don't understand. 
If it is local date, so you need to parse it, not only add to it "Z"

Reply:
No no, I don't change my date, I changed the XSOAR date,
For example:
CREATEDATE: 2022-03-28T:14:00:00 server in Istanbul.
But at this time XSOAR time is 2022-03-28T:11:00:00 in UTC.
So when I compare the dates I add XSOAR time the utc difference. For difference I added instance parameter.

Reply:
Why not to change your time to UTC from start when you get it?

Reply:
But you are not changing XSOAR time to UTC, you are changing your time to UTC (in not a good way):
`time_stamp = inc['CorrelationAlert']['CREATEDATE'] + "Z"`

Reply:
No, inc['CorrelationAlert']['CREATEDATE'] give local time but not completly iso_formatted. I added 'Z'  to tag its' utc times but not utc+0.
if time_stamp is 16:00:00 for my product , this mean is 13:00:00 for utc.
So, I added `timedelta(hours=int(demisto.params().get("time_zone_difference",3))` to XSOAR times.
time_zone_difference is taken from instance params.